wire/internal/wire/testdata/MultipleBindings/foo/wire.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//+build wireinject
+
+package main
+
+import (
+	"strings"
+
+	"github.com/google/go-cloud/wire"
+)
+
+func inject() Foo {
+	// fail: provideFoo and provideFooAgain both provide Foo.
+	panic(wire.Build(provideFoo, provideFooAgain))
+}
+
+func injectFromSet() Foo {
+	// fail: provideFoo is also provided by Set.
+	panic(wire.Build(provideFoo, Set))
+}
+
+func injectFromNestedSet() Foo {
+	// fail: provideFoo is also provided by SuperSet, via Set.
+	panic(wire.Build(provideFoo, SuperSet))
+}
+
+func injectFromSetWithDuplicateBindings() Foo {
+	// fail: DuplicateBindingsSet has two providers for Foo.
+	panic(wire.Build(SetWithDuplicateBindings))
+}
+
+func injectDuplicateValues() Foo {
+	// fail: provideFoo and wire.Value both provide Foo.
+	panic(wire.Build(provideFoo, wire.Value(Foo("foo"))))
+}
+
+func injectDuplicateInterface() Bar {
+	// fail: provideBar and wire.Bind both provide Bar.
+	panic(wire.Build(provideBar, wire.Bind(new(Bar), strings.NewReader("hello"))))
+}
```

Comment:
Anything else I should add? I considered adding a struct provider, do you think that's necessary?

Reply:
I think this covers sufficiently. If you want to add it, go for it, but I don't see it as required.